lib/IDE/SwiftSourceDocInfo.cpp

```diff
@@ -342,6 +347,22 @@ struct RangeResolver::Implementation {
     std::vector<ASTNode> EndMatches;
     ContextInfo(ASTNode Parent, bool ContainedInRange) : Parent(Parent),
       ContainedInRange(ContainedInRange) {}
+
+    bool isMultiStatment() {
+      if (StartMatches.empty() || EndMatches.empty())
+        return false;
+
+      // Multi-statement should have a common parent of brace statement, this
+      // can be implicit brace statement, e.g. in case statement.
+      if (Parent.isStmt(StmtKind::Brace))
```

Comment:
The is implied in `ContextInfo`; they won't end up here if they are not siblings.